cypress/integration/x-msc2716-historical-import/historical-import.spec.ts

```diff
@@ -0,0 +1,288 @@
+    });
+}
+
+function ensureVirtualUsersRegistered(synapse: SynapseInstance, applicationServiceToken, virtualUserIds: string[]) {
+    const url = `${synapse.baseUrl}/_matrix/client/r0/register`;
+
+    const virtualUserLocalparts = virtualUserIds.map((virtualUserId) => {
+        const userIdWithoutServer = virtualUserId.split(':')[0];
+        const localpart = userIdWithoutServer.replace(/^@/, '');
+        return localpart;
+    });
+
+    virtualUserLocalparts.forEach((virtualUserLocalpart) => {
+        cy.request<{ error?: string, errcode?: string }>({
+            url,
+            method: "POST",
+            body: {
+                type: "m.login.application_service",
+                username: virtualUserLocalpart
+            },
+            headers: {
+                'Authorization': `Bearer ${applicationServiceToken}`
+            },
+            // We'll handle the errors ourselves below
+            failOnStatusCode: false,
+        })
+            .then((res) => {
+                // Registration success
+                if(res.status === 200) {
+                    return;
+                }
+
+                const errcode = res.body.errcode;
+
+                // User already registered and good to go
+                if (res.status == 400 && errcode === "M_USER_IN_USE") {
+                    return;
+                }
+
+                const errorMessage = res.body.error;
+                throw new Error(`ensureVirtualUserRegistered failed to register: (${errcode}) ${errorMessage}`)
+            });
+    });
+}
+
+describe("MSC2716: Historical Import", () => {
+    let synapse: SynapseInstance;
+    let asMatrixClient: MatrixClient;
+    let loggedInUserCreds: UserCredentials;
+    const virtualUserIDs = ['@maria-01234:localhost'];
+
+    // This corresponds to the application service registration defined in the
+    // "msc2716-historical-import" Synapse configuration
+    const AS_TOKEN = 'as_token123';
+
+    beforeEach(() => {
+        cy.window().then(win => {
+            // Collapse left panel for these tests (get more space in the area we care about)
+            win.localStorage.setItem("mx_lhs_size", "0");
+        });
+        // Start Synapse with msc2716_enabled and an application service configured
+        cy.startSynapse("msc2716-historical-import").then(data => {
+            synapse = data;
+
+            // This is the person we're logged in as
+            cy.initTestUser(synapse, "Grace").then(userCreds => {
+                loggedInUserCreds = userCreds;
+            });
+            // After the page is loaded from initializing the logged in user so
+            // the mxSettingsStore is available
+            cy.window().then(win => {
+                // Disable the sound notifications so you're not startled and
+                // confused where the sound is coming from
+                win.mxSettingsStore.setValue("audioNotificationsEnabled", null, SettingLevel.DEVICE, false);
```

Comment:
How do I fix this TypeScript error? This does work functionally.

> Property 'setValue' does not exist on type 'SettingsStore'. Did you mean to access the static member 'SettingsStore.setValue' instead?

We can't just import `SettingsStore` at the top of the file and do `SettingsStore.setValue(...)` because it's not in the right context and Cypress doesn't understand TypeScript anyway.

Reply:
Exactly like you're doing it but it has to be described in the TS definition of the AUTWindow

Reply:
For reference, waiting to resolve this until we get JSX support, https://github.com/matrix-org/matrix-react-sdk/pull/8354#discussion_r883260722